modules/compute/gke-node-pool/main.tf

```diff
@@ -253,3 +253,98 @@ resource "google_project_iam_member" "node_service_account_artifact_registry" {
   role    = "roles/artifactregistry.reader"
   member  = "serviceAccount:${local.sa_email}"
 }
+
+# Enable GPUDirect for A3 and A3Mega VMs, this involve multiple kubectl steps to integrate with the created cluster
+# 1. Install NCCL plugin daemonset
+# 2. Install NRI plugin daemonset
+# 3. Update user workload to inject rxdm sidecar and other required annotation, volume etc.
+locals {
+  split_cluster_id         = split("/", var.cluster_id)
+  user_workload_path_tcpx  = var.user_workload_path == null ? "${path.module}/gpu-direct-workload/sample-tcpx-workload-job.yaml" : var.user_workload_path
+  user_workload_path_tcpxo = var.user_workload_path == null ? "${path.module}/gpu-direct-workload/sample-tcpxo-workload-job.yaml" : var.user_workload_path
+
+  # Manifest to be installed for enabling TCPX on A3 machines
+  tcpx_manifests = [
+    "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/gpudirect-tcpx/nccl-tcpx-installer.yaml",      # nccl_plugin v3.1.9 for tcpx
+    "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/gpudirect-tcpx/nccl-config.yaml",              # nccl_configmap
+    "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/nri_device_injector/nri-device-injector.yaml", # nri_plugin
+  ]
+
+  # Manifest to be installed for enabling TCPXO on A3Mega machines
+  tcpxo_manifests = [
+    "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/gpudirect-tcpxo/nccl-tcpxo-installer.yaml",    # nccl_plugin v1.0.4 for tcpxo
+    "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/nri_device_injector/nri-device-injector.yaml", # nri_plugin
+  ]
+
+  updated_user_workload_path = (
```

Comment:
- Maybe the script can accept exported file path/name as an input?
-  Or the script can return the content of the manipulated yaml as a string and we apply it on the fly without saving the file? This can be a topic to discuss if we have to save/keep the manipulated file.

Reply:
we discussed this in the previous meeting. the toolkit focus on setting up the cluster/np and it is up to user to decide whether/when they will deploy a sample workload to it or not.
Further more, outputting the updated file is easier for user to visualized and debug in the long run.